sale_coupon_multi_gift/models/sale_coupon_program.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2021 Tecnativa - David Vidal
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import models
+
+
+class SaleCouponProgram(models.Model):
+    _inherit = "sale.coupon.program"
+
+    def _compute_order_count(self):
+        """Relay on the order line link for these programs instead of the discount
+        products"""
+        multi_gift_programs = self.filtered(lambda x: x.reward_type == "multi_gift")
+        super(SaleCouponProgram, self - multi_gift_programs)._compute_order_count()
+        for program in multi_gift_programs:
+            orders = self.env["sale.order.line"].read_group(
+                [
+                    ("state", "not in", ["draft", "sent", "cancel"]),
+                    ("coupon_program_id", "=", program.id),
+                ],
+                ["order_id"],
+                ["order_id"],
+            )
+            program.order_count = len(orders)
```

Comment:
This is not correct. You should group by `coupon_program_id` and `order_id`, with `lazy=False`, and unfold the count.

Reply:
In this particular case that's no valid, as one promotion unfolds several order lines, so we'd obtain a count equal to the number of rewarded products for each order.

This way, I just count the dicts (one per order) of the result to obtain the count I need (number of orders, not number of rewards)

![image](https://user-images.githubusercontent.com/5040182/127275096-5dab344d-6d0d-4379-a365-c698e2e46074.png)